packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/arrow.dart

```diff
@@ -0,0 +1,280 @@
+            headSize: arrowHeadSize,
+            strokeWidth: arrowStrokeWidth,
+            type: ArrowType.down,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+/// Widget that draws a fully sized, centered, unidirectional arrow according to its constraints
+@immutable
+class ArrowWidget extends StatelessWidget {
+  const ArrowWidget({
+    this.color = Colors.white,
+    this.headSize = 16.0,
+    Key key,
+    this.strokeWidth = 2.0,
+    @required this.type,
+  })  : assert(color != null),
+        assert(headSize != null),
+        assert(strokeWidth != null),
+        assert(type != null),
+        super(key: key);
+
+  final Color color;
+
+  /// The arrow head is a Equilateral triangle
+  final double headSize;
+
+  final double strokeWidth;
+
+  final ArrowType type;
+
+  CustomPainter get _painter => _ArrowPainter(
+        arrowHeadSize: headSize,
+        color: color,
+        strokeWidth: strokeWidth,
+        strategy: _ArrowPaintStrategy.strategy(type),
+      );
+
+  @override
+  Widget build(BuildContext context) {
+    return CustomPaint(
+      painter: _painter,
+      child: Container(),
+    );
+  }
+}
+
+abstract class _ArrowPaintStrategy {
+  void paint({
+    @required Canvas canvas,
+    @required Size size,
+    @required Paint paint,
+    @required double headSize,
+  });
+
+  static final _ArrowPaintStrategy _up = _UpwardsArrowPaintStrategy();
+  static final _ArrowPaintStrategy _left = _LeftwardsArrowPaintStrategy();
+  static final _ArrowPaintStrategy _down = _DownwardsArrowPaintStrategy();
+  static final _ArrowPaintStrategy _right = _RightwardsArrowPaintStrategy();
+
+  static Path pathForArrowHead(Offset p1, Offset p2, Offset p3) {
+    return Path()
+      ..moveTo(p1.dx, p1.dy)
+      ..lineTo(p2.dx, p2.dy)
+      ..lineTo(p3.dx, p3.dy)
+      ..close();
+  }
+
+  static _ArrowPaintStrategy strategy(ArrowType type) {
+    switch (type) {
+      case ArrowType.up:
+        return _up;
+      case ArrowType.left:
+        return _left;
+      case ArrowType.right:
+        return _right;
+      case ArrowType.down:
+        return _down;
+      default:
+        return _up;
+    }
+  }
+}
+
+class _UpwardsArrowPaintStrategy implements _ArrowPaintStrategy {
+  @override
+  void paint({Canvas canvas, Size size, Paint paint, double headSize}) {
+    final arrowHeadSizeDividedByTwo = headSize / 2;
+    final p1 = Offset.zero;
+    final p2 = Offset(-arrowHeadSizeDividedByTwo, headSize);
+    final p3 = Offset(arrowHeadSizeDividedByTwo, headSize);
+    canvas.drawPath(_ArrowPaintStrategy.pathForArrowHead(p1, p2, p3), paint);
+    final lineStartingPoint = Offset(0, headSize);
+    final lineEndingPoint = Offset(0, size.height);
+    canvas.drawLine(lineStartingPoint, lineEndingPoint, paint);
+  }
+}
+
+class _LeftwardsArrowPaintStrategy implements _ArrowPaintStrategy {
+  @override
+  void paint({Canvas canvas, Size size, Paint paint, double headSize}) {
+    final arrowHeadSizeDividedByTwo = headSize / 2;
```

Comment:
there is a lot of overlap between these 4 strategy methods. Add a helper method to reduce the duplicate code.
I expect you can add a 
paintArrow(lineStartingPoint, lineEndingPoint, paint); that handles all 4 orientations you care about.

Reply:
One special case you will want to handle is if the line isn't long enough to support drawing the full head. For example, if the distance between lineStartingPoint and lineEndingPoint is less than headSize.